src/network/serverpackethandler.cpp

```diff
@@ -1050,12 +1071,16 @@ void Server::handleCommand_Interact(NetworkPacket *pkt)
 		}
 		float d = playersao->getEyePosition().getDistanceFrom(target_pos);
 
-		if (!checkInteractDistance(player, d, pointed.dump())
-				&& pointed.type == POINTEDTHING_NODE) {
-			// Re-send block to revert change on client-side
-			RemoteClient *client = getClient(peer_id);
-			v3s16 blockpos = getNodeBlockPos(pointed.node_undersurface);
-			client->SetBlockNotSent(blockpos);
+		const auto what = pointed.dump();
+		bool isOk = checkInteractDistance(player, d, what) &&
```

Comment:
Is there a reason why the interaction distance was only applied to nodes before?

Reply:
I cannot imagine any reason. Though, the allowed distance should be increased because this function assumes a maximal selection box of 1.5m into each direction, though entities can have larger ones.